### DISCUSSION TITLE
Website Deployment

### DISCUSSION THREAD

**Discussion by sekwah41** [zidoro/pomatez#225] [General]
@roldanjr what are you currently using to deploy the gh-pages branch or are you manually committing it atm?

I can also sort that out to be automated too if you'd like.

**Comment by roldanjr** [comment]
> I can also sort that out to be automated too if you'd like.

Currently, I manually deploy it to gh-pages using my computer.
I cannot deploy it manually right now due to some errors on my machine.

Yeah, I think it's better if there is some sort of automation for the deployment of the website.
I was planning to host it on Vercel and use NextJS instead of Gatsby when the time allows me to do it.

What do you think @sekwah41 .

**Reply by sekwah41** [reply]
Wait why would you host it on Vercel? Couldn't you just host it on Github pages with a custom dns if you are wanting that? Its what I do for my Minecraft plugin https://advancedportals.sekwah.com/docs/commands

Or is Vercel a framework? I've not used either of those directly before but I've heard of Gatsby.

**Reply by roldanjr** [reply]
The framework I used to built this website is [GatsbyJS](https://www.gatsbyjs.com/). But I find it very dependent on third-party plugins. Though it has some advantages also.

Most of the websites I built nowadays are built using [NextJS](https://nextjs.org/) framework and I hosted it on Vercel.
Every time there are some changes I made,  Vercel kinda deploys it right away.

I will still think of it if it's worth the effort to do so.

But yeah, If you can automate the deployment of the current website it's better than revamping and transitioning to a different hosting provider.

Thank you in advance @sekwah41 🙇🏼‍♂️

**Reply by gillisandrew** [reply]
Having run into random errors in old gatsby projects, it's almost certainly an issue created by updating nodejs on your machine. Try running it with an older node (docker or through nvm). That said we can use github actions to automate the build and deploy the site without adopting nextjs.

**Comment by gillisandrew** [comment]
Didn't take too long to bring the webpage down 😆 

Let me know if you need a hand getting that github action working haha

**Reply by sekwah41** [reply]
The annoying part is working in the dev stuff though seems to be compiling different in the build stage. Will output the artifacts and do some tests after reverting the gh-page branch.

**Reply by gillisandrew** [reply]
It's because you aren't passing `--prefix-paths` as a build argument. You actually need to double escape them to do it from the root `package.json`.

That makes the build command:
```
yarn build:site -- -- --prefix-paths
```

**Reply by gillisandrew** [reply]
There are other issues with the workflow but I think that should be enough to get the site up again.

**Reply by sekwah41** [reply]
Ahh im looking now and reason it worked with manual deployments ie because the predeploy command has already got that variable set.

**Reply by sekwah41** [reply]
> There are other issues with the workflow but I think that should be enough to get the site up again.

I've already pushed a revert commit though I'll do some more checks locally to see that its building with the right links.

**Reply by sekwah41** [reply]
Ah wait the issue is that the website is built against / as the root rather than pomatez/ usually I end up having this all configured in the config files. Usually in the libraries I use its just solely set in a config value.

**Reply by gillisandrew** [reply]
Its set in the config values [here](https://github.com/roldanjr/pomatez/blob/master/website/src/config.js)

Having had a closer look, a bunch of environment variables have to be set so maybe just hold off on deploy until we can get it properly configured.

**Reply by sekwah41** [reply]
Seems gatsby ignores it without the --prefix-paths which is what I meant.

What other env variables are needing to be set for now? At least locally to get back to the same state as the the currently deployed website it can be done using PREFIX_PATHS=true
